### ISSUE TITLE
Forge 31.2.0 breaks Java 11 compatibility

### ISSUE THREAD

**Issue by pluto00987** [MinecraftForge/MinecraftForge#6710]
**Minecraft Version:** 1.15.2

**Forge Version:** 31.2.0

**Logs:** https://gist.github.com/pluto00987/100fe9ae6308a3275df5a801e7bb2e23

**Description of issue:**
Crash on load when using Java jdk/jre 11.0.7 (also 11.0.6). No mods.
Works OK with jre 1.8.0_251, and also 14, 14.0.1

Jdk 11.0.x works OK with Forge 31.1.0 but 13 and up did not. See also issue #6576

**Comment by gigaherz** [comment]
I can't reproduce the issue. Works fine with my openjdk 11.0.6 
![image](https://user-images.githubusercontent.com/685851/82623629-e9759c00-9be0-11ea-905d-529bb181e233.png)

**Comment by pluto00987** [comment]
Openjdk, must be linux? I didn't actually state it but I'm on windows (7) using the 'real' oracle jdk. Fresh install of forge 31.2.0, changing nothing but the java path in the launcher. Maybe some subtle difference between JDKs? There's no openjdk 11.0.6 available for windows to test. I'll see if I can find anything else to try..

**Comment by gigaherz** [comment]
Actually, I'm on Windows 10. OpenJDK doesn't imply Linux anymore. The JDK and JRE downloads from Oracle have a new license now, which makes most of the previous uses of the Oracle versions of Java disallowed without a paid license. They themselves recommend OpenJDK to most people, since the OpenJDK builds such as those from the [AdoptOpenJDK project](https://adoptopenjdk.net/) now contain everything you'd ever need to use Java.

**Comment by FinnT730** [comment]
Could it be that it might be windows 7?

**Comment by LexManos** [comment]
Report any and all of these types of errors to https://github.com/jhalterman/typetools/pull/66
We've patched as much as we can and tested what we can. 
But beyond that, use the recommended JDKs and you wont have a issue. I've personally tested windows 10, JDK 8-11 both orical and openjdk. So they should be fine.

**Comment by pluto00987** [comment]
Whoops, forgot about this... Just for completeness I'll add these comments I had intended to post but never did:

@gigaherz Oracle Java is still available for free for (among other things) "personal use on a desktop or laptop computer, such as to play games or run other personal applications." [(license FAQ)](https://www.oracle.com/technetwork/java/javase/overview/oracle-jdk-faqs.html) You do have to sign up for a (free) OTN developer account in order to download it though, so it's a bit of hassle. I wasn't actually aware of AdoptOpenJDK having windows builds. TBH I'll probably start using that going forward.

In regards to this issue, I did download AdoptOpenJDK 11.0.7+10 and it works with forge 31.2.0 just fine for me. The Oracle JDK 11.0.7 still yields the crash I reported however. Somewhat surprising to see what appears to be a JDK-specific fault but I guess that's what it is. Perhaps related to Win 7 specifically. I would've though that unlikely too but who knows?

Thanks!
